src/raven/client.clj

```diff
@@ -385,6 +387,38 @@
     :data data
     :env env}))
 
+(defn get-full-ring-url
+  "Given a ring compliant request object, return the full URL.
+    This was lifted from ring's source code so as to not depend on it."
+  [request]
+  (str (-> request :scheme name)
+       "://"
+       (get-in request [:headers "host"])
```

Comment:
`port` ?

Reply:
https://github.com/ring-clojure/ring/blob/master/SPEC

Reply:
As you can see in the tests, the "host" header includes the port.

This function was taken as is from https://github.com/ring-clojure/ring/blob/master/ring-core/src/ring/util/request.clj#L5 , as per the comment string.

Does your comment imply a functional change is needed? I'm not sure what to do.

Reply:
right! we'll see as we go if it needs to use the `x-forwarded-for` header.